consensus-types/forks/eip4844/tx.go

```diff
@@ -16,16 +16,65 @@ var (
 	errInvalidNumBlobVersionedHashes = errors.New("invalid number of blob versioned hashes")
 )
 
+const (
+	versionedHashesOffset = 258 // offset the versioned_hash_offset in a serialized blob tx
+	messageLen            = 192 // size in bytes of "message" within the serialized blob tx
+)
+
+// TxPeekBlobVersionedHashes is from EIP-4844, and extracts the list of versioned hashes from the
+// given blob tx.
+//
+// Format of the blob tx relevant to this function is as follows:
+//   0: type (value should always be BlobTxType, 1 byte)
+//   1: message offset (value should always be 69, 4 bytes)
+//   5: ECDSA signature (65 bytes)
+//   70: start of "message" (192 bytes)
+//     258: start of the versioned hash offset within "message"  (4 bytes)
+//   262-: rest of the tx following message
+//
+// TODO: unit tests, remove dependency on github.com/protolambda/ztyp
```

Comment:
btw, you can test using:
```
bazel test //beacon-chain/sync:go_default_test --test_filter='TestValidateBeaconBlockPubSub_ValidBlobKzgs'
```

Reply:
though a specific unit test for this function would also be desired.

Reply:
thanks! The test seems to also rely on geth so it fails until I point it at my local geth repo, which I guess confirms it's WAI!